### PR TITLE
Add basic support for cloud devices

### PR DESCRIPTION
**Pull request by rowysock**

### CODE REVIEW

File: custom_components/homewhiz/config_flow.py [hunk 1]
```diff
@@ -45,55 +53,60 @@ class TiltConfigFlow(ConfigFlow, domain=DOMAIN):
...
+        self._bt_name: str | None = None
+        self._cloud_config: CloudConfig | None = None
+        self._cloud_credentials: LoginResponse | None = None
+        self._cloud_appliances: list[ApplianceInfo] | None = None
 
     async def async_step_bluetooth(
         self, discovery_info: BluetoothServiceInfoBleak
     ) -> FlowResult:
         """Handle the bluetooth discovery step."""
+        print("BT!!", discovery_info)
```
Comment (TechHummel) [inline custom_components/homewhiz/config_flow.py:67]: logging? 😄

File: custom_components/homewhiz/config_flow.py [hunk 2]
```diff
@@ -45,55 +53,60 @@ class TiltConfigFlow(ConfigFlow, domain=DOMAIN):
...
-        self._discovery_info = discovery_info
-        self._address = discovery_info.address
-        self._name = discovery_info.name
-        return await self.async_step_credentials()
+        self._bt_address = discovery_info.address
+        self._bt_name = discovery_info.name
+        return await self.async_step_bluetooth_connect()
+
+    async def async_step_user(self, user_input: dict[str, Any] | None = None):
+        print("USER!!")
```
Comment (TechHummel) [inline custom_components/homewhiz/config_flow.py:78]: Same as before

File: custom_components/homewhiz/api.py [hunk 3]
```diff
@@ -213,54 +217,80 @@ async def make_api_get_request(
...
         credentials=credentials,
     )
-    contentsIndex = from_dict(ContentsIndexResponse, contents_index_response["data"])
-    config_contents = [
-        content for content in contentsIndex.results if content.ctype == "CONFIGURATION"
-    ]
+    return from_dict(ContentsIndexResponse, response["data"])
+
+
+async def fetch_base_contents_index(credentials: LoginResponse, language: str):
```
Comment (TechHummel) [inline custom_components/homewhiz/api.py:237]: Doesn't really matter but `fetch_appliance_contents` has a default language while this function doesn't.

File: custom_components/homewhiz/config_flow.py (hunk 1 above)
Comment (rowysock) [reply custom_components/homewhiz/config_flow.py]: removed

File: custom_components/homewhiz/config_flow.py (hunk 2 above)
Comment (rowysock) [reply custom_components/homewhiz/config_flow.py]: removed

File: custom_components/homewhiz/api.py (hunk 3 above)
Comment (rowysock) [reply custom_components/homewhiz/api.py:237]: I actually removed it on purpose, as this one's purpose is to fetch only the localizations and should never be called without the language. Maybe I should rename it?

Comment (TechHummel) [reply custom_components/homewhiz/api.py:237]: No, makes sense!

**Review by TechHummel — APPROVED** [review]

**Pull request merged**